gapis/gfxapi/vulkan/dependency_graph.go

```diff
@@ -0,0 +1,1733 @@
+		framebuffer := beginInfo.Framebuffer
+		addRead(&b, g, vulkanStateKey(framebuffer))
+		renderpass := beginInfo.RenderPass
+		addRead(&b, g, vulkanStateKey(renderpass))
+
+		atts := GetState(s).Framebuffers.Get(framebuffer).ImageAttachments
+		attDescs := GetState(s).RenderPasses.Get(renderpass).AttachmentDescriptions
+		for i := uint32(0); i < uint32(len(atts)); i++ {
+			img := atts.Get(i).Image.VulkanHandle
+			loadOp := attDescs.Get(i).LoadOp
+			if loadOp == VkAttachmentLoadOp_VK_ATTACHMENT_LOAD_OP_CLEAR {
+				// This can be wrong as this is clearing all the memory bindings
+				// that overlap with the attachment image, so extra memories might be
+				// cleared. However in practical, image should be bound to only one
+				// memory binding as a whole. So here should be a problem.
+				// TODO: Add intersection operation to get the memory ranges to clear
+				bindingsToClear := getOverlappedBindingsForImage(img)
+				addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+				addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+				recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+					writeMemoryBindingsData(b, bindingsToClear)
+				})
+			} else {
+				bindingsToRead := getOverlappedBindingsForImage(img)
+				addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+				addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+				recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+					readMemoryBindingsData(b, bindingsToRead)
+				})
+			}
+		}
+
+	case *RecreateCmdBeginRenderPass:
+		beginInfo := typedAtom.PRenderPassBegin.Read(ctx, a, s, nil)
+		framebuffer := beginInfo.Framebuffer
+		addRead(&b, g, vulkanStateKey(framebuffer))
+		renderpass := beginInfo.RenderPass
+		addRead(&b, g, vulkanStateKey(renderpass))
+
+		atts := GetState(s).Framebuffers.Get(framebuffer).ImageAttachments
+		attDescs := GetState(s).RenderPasses.Get(renderpass).AttachmentDescriptions
+		for i := uint32(0); i < uint32(len(atts)); i++ {
+			img := atts.Get(i).Image.VulkanHandle
+			loadOp := attDescs.Get(i).LoadOp
+			if loadOp == VkAttachmentLoadOp_VK_ATTACHMENT_LOAD_OP_CLEAR {
+				// This can be wrong as this is clearing all the memory bindings
+				// that overlap with the attachment image, so extra memorise might be
+				// cleared. However in real case, image should be bound to only one
+				// memory binding as a whole. So here should be cause problem.
+				// TODO: Add intersection operation to get the memory ranges to clear
+				bindingsToClear := getOverlappedBindingsForImage(img)
+				addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+				addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+				recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+					writeMemoryBindingsData(b, bindingsToClear)
+				})
+			} else {
```

Comment:
Added. However, if the storeOp is DONT_CARE, all the 'modify' and 'write' in that renderpass should be disabled. This is not implemented. TODO is added there.